api/src/main/java/ca/bc/gov/educ/api/distribution/process/PSIReportProcess.java

```diff
@@ -405,6 +429,50 @@ private void writesCsvFileRowA(List<String[]> studentTranscriptdata, String pen,
 
     }
 
+    private static int getProgramCodesListSize(Student studentDetails) {
+        List<String> optionalOrCareerProgramCodes = studentDetails.getGraduationData().getProgramCodes();
+        return optionalOrCareerProgramCodes != null ? optionalOrCareerProgramCodes.size() : 0;
+    }
+
+    private static String getHonorsFlag(Student studentDetails) {
+        return String.valueOf(studentDetails.getGraduationData().getHonorsFlag()).isBlank() ? "" : String.valueOf(studentDetails.getGraduationData().getHonorsFlag());
+    }
+
+    private static String getDogWoodFlag(Student studentDetails) {
+        return String.valueOf(studentDetails.getGraduationData().getDogwoodFlag()).isBlank() ? "" : String.valueOf(studentDetails.getGraduationData().getDogwoodFlag());
+    }
+
+    private static String getNonGradReasons(List<NonGradReason> nonGR) {
+        return ((nonGR == null) || nonGR.isEmpty()) ? "" : nonGR.stream()
+                .map(NonGradReason::getCode)
+                .filter(Objects::nonNull)
+                .limit(15)
+                .collect(Collectors.joining(""));
+    }
+
+    private static String getProgramCompleteionDate(Student studentDetails) {
```

Comment:
Minor spelling: getProgramCompletionDate instead of getProgramCompleteionDate

Reply:
@cditcher Corrected the spelling.